cmake/pristine.cmake

```
@@ -32,7 +32,7 @@ if(NOT INDEX EQUAL -1)
   message(FATAL_ERROR "Refusing to run pristine in in-source build folder.")
 endif()
 
-file(GLOB build_dir_contents ${BINARY_DIR}/*)
+file(GLOB build_dir_contents LIST_DIRECTORIES TRUE ${BINARY_DIR}/*)
```

Would something like this also work? Potentially checking for the existence of `${BINARY_DIR}` first?
```suggestion
file(REMOVE_RECURSE ${BINARY_DIR}/)
```

---

Bump

---

Yeah, I think my preference would be to keep it as simple as possible.

Probably @tejlmand will have some important feedback as well.

---

This change should not be needed.
Can you provide more details regarding the actual issue seen on your system ?

`file(GLOB ...)` should return directories by default, and that's also the behavior I see on all machines I tried this on.
From the docs:
> Added in version 3.3: By default `GLOB` lists directories. Directories are omitted in the result if `LIST_DIRECTORIES` is set to false.

Ref: https://cmake.org/cmake/help/latest/command/file.html#filesystem

Note, behavior is different for `GLOB_RECURSE`

---

> Anyway sorry to bother.

@nlebedenco No worries.
And thanks for raising attention to a potential issue, even if it potentially is a CMake one.

If you experience the original issue again on Windows, the feel free to open a bug with detailed description, cause even if it is CMake internal error then we might have ways to work around the issue using other functions.